### ISSUE TITLE
Misleading description of poetry

### ISSUE THREAD

**Issue by sdispater** [pypa/python-packaging-user-guide#506]
> poetry for Python developer focused components that are designed primarily for publication to a Python package index (pipenv deliberately avoids making the assumption that the application being worked on will support distribution as a pip-installable Python package, while poetry based applications rely explicitly on their Python packaging metadata to describe their application structure and dependencies)

As the author of Poetry, I find this somewhat misleading.

Like I said on Twitter, there is a misconception around Poetry and what its purpose is. Poetry is a tool to manage Python projects, being an application or a library.

> poetry based applications rely explicitly on their Python packaging metadata

This is not true, the only metadata necessary in Poetry are the project name, its version and one or more author. That’s it. I made this choice because for me every project should have a name and a version.

The only difference when managing an application compared to a library is that the lock file should be check into version control and you won’t have to use the build and publish command. Nothing more.

Basically, the workflow you have with pipenv can be replicated with poetry:

## Installing packages for your project
```
poetry add requests
```

This displays an output like this:

```
Using version ^2.18 for requests

Updating dependencies
Resolving dependencies... (0.1s)


Package operations: 5 installs, 0 updates, 0 removals

Writing lock file

  - Installing certifi (2018.4.16)
  - Installing chardet (3.0.4)
  - Installing idna (2.6)
  - Installing urllib3 (1.22)
  - Installing requests (2.18.4)
```

## Using installed packages

```
poetry run python main.py
```

So it’s seems that this was written by someone who has not used Poetry and has preconceptions about what it does or not. So I thought I would try to clarify things a bit.

**Comment by theacodes** [comment]
Thanks, @sdispater - do you have any specific wording you'd like to suggest that we use when linking to poetry?

**Comment by ncoghlan** [comment]
The config file for poetry is `pyproject.toml` - that's a Python-specific packaging metadata format, and the main reason to use it is if you're wanting to interoperate with other tools in the Python ecosystem that know how to read it.

The reason `pipenv` deliberately doesn't use `pyproject.toml` is because we explicitly want to separate the "PyPI consumer" case (where components are consumed from PyPI, either for publication through other means, or not for publication at all, but the project as a whole doesn't conform to the requirements of a Python sdist), from the "PyPI publisher" case (where projects are structured in a way that conforms to either the legacy sdist format or the newer one defined in PEPs 517 & 518, and hence can be built automatically by Python-specific build tools).

Historically that distinction was framed as `requirements.txt` vs `setup.py`, now its framed as `Pipfile` vs `pyproject.toml`. But it's not an accident, and it's not an oversight - it's a deliberate choice.

The folks most likely to find that choice inconvenient are those that would prefer to define everything in `pyproject.toml` and not have any other config files, which is why the reference out to `poetry` at the end of the tutorial is framed the way it is: it's not the entirety of the potential audience for `poetry`, but it's the portion of that audience that are least likely to be happy with the design decisions made in `pipenv`.

**Comment by sdispater** [comment]
> If you look at a pyproject.toml like:
>
>[tool.poetry.dependencies]
>cleo = { git = "https://github.com/sdispater/cleo.git", branch = "master" }
>
>Having that git dependency there makes this something that can't be uploaded to PyPI, because that points to a concrete location and dependencies inside of a package are only abstract until they're fully resolved.

That's why if you want to depend on a git dependency in development you would put it in the `dev-dependencies` section.

```toml
[tool.poetry.dependencies]
cleo = "^0.6.6"

[tool.poetry.dev-dependencies]
cleo = { git = "https://github.com/sdispater/cleo.git", branch = "master" }
```

That way, you can have concrete dependencies and abstract dependencies all in one file.

**Comment by ncoghlan** [comment]
Thanks for clarifying your concerns with the current reference wording @sdispater - I'll put together a PR that hopefully clarifies things, and we can iterate on some specific wording there.

@dstufft Agreed on the fact that we could have nested `Pipfile` inside `pyproject.toml` at a technical level (putting aside the question of when the respective formats were invented), and that the core semantic distinction is between "I am the system integrator for these dependencies" (concrete dependencies) and "I am a component publisher declaring known constraints that someone *else* will need to decide how to satisfy at integration time" (abstract dependencies).

**Comment by ncoghlan** [comment]
#508 is the PR with the rewording. The commit message is longer than the docs are, and it does focus on the file structure, as that seems to be what folks sometimes find most counter-intuitive about the `pipenv` model (i.e. the fact that if you have a `pipenv`-managed repository for a PEP 517/518 based repo, you'll end up with both `Pipfile` and `pyproject.toml` at the top level).

While concrete vs abstract dependencies is the explanation for *why* we have that split, I don't think it's why folks sometimes object to having two files: I think that's because for their particular use case, they're in a situation where they don't lose any clarity by dropping back to `pyproject.toml` as their sole top level dependency declaration file.
